homeassistant/components/insteon/config_flow.py

```diff
@@ -0,0 +1,353 @@
+"""Test config flow for Insteon."""
+import logging
+
+from pyinsteon import async_connect
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_ADDRESS,
+    CONF_DEVICE,
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+# pylint: disable=unused-import
+from .const import (
+    CONF_HOUSECODE,
+    CONF_HUB_VERSION,
+    CONF_OVERRIDE,
+    CONF_UNITCODE,
+    CONF_X10,
+    DOMAIN,
+    SIGNAL_ADD_DEVICE_OVERRIDE,
+    SIGNAL_ADD_X10_DEVICE,
+    SIGNAL_REMOVE_DEVICE_OVERRIDE,
+    SIGNAL_REMOVE_X10_DEVICE,
+)
+from .schemas import (
+    add_device_override,
+    add_x10_device,
+    build_device_override_schema,
+    build_hub_v1_schema,
+    build_hub_v2_schema,
+    build_plm_schema,
+    build_remove_override_schema,
+    build_remove_x10_schema,
+    build_x10_schema,
+)
+
+STEP_PLM = "plm"
+STEP_HUB_V1 = "hubv1"
+STEP_HUB_V2 = "hubv2"
+STEP_CHANGE_HUB_CONFIG = "change_hub_config"
+STEP_ADD_X10 = "add_x10"
+STEP_ADD_OVERRIDE = "add_override"
+STEP_REMOVE_OVERRIDE = "remove_override"
+STEP_REMOVE_X10 = "remove_x10"
+MODEM_TYPE = "modem_type"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _only_one_selected(*args):
+    """Test if only one item is True."""
+    return sum(args) == 1
+
+
+async def _async_connect(**kwargs):
+    """Connect to the Insteon modem."""
+    try:
+        await async_connect(
+            device=kwargs.get(CONF_DEVICE),
+            host=kwargs.get(CONF_HOST),
+            port=kwargs.get(CONF_PORT),
+            username=kwargs.get(CONF_USERNAME),
+            password=kwargs.get(CONF_PASSWORD),
+            hub_version=kwargs.get(CONF_HUB_VERSION),
```

Comment:
Any of these are are required should use `[]`s

Reply:
The configuration can be set up with either the PLM or the HUB. So technically none of these are always "required" but you need either 1. `CONF_DEVICE` or 2. `CONF_HOST`, but not both.  That is all handled in the schema so there is no concern to check the validity here. The only question here is which set of configuration is used.